resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -368,14 +364,11 @@ private[spark] class KubernetesClusterSchedulerBackend(
    *         locality if an executor launches on the cluster node.
    */
   private def getNodesWithLocalTaskCounts() : Map[String, Int] = {
-    val executorPodsWithIPs = EXECUTOR_PODS_BY_IPS_LOCK.synchronized {
-      executorPodsByIPs.values.toList  // toList makes a defensive copy.
-    }
     val nodeToLocalTaskCount = mutable.Map[String, Int]() ++
       KubernetesClusterSchedulerBackend.this.synchronized {
         hostToLocalTaskCount
       }
-    for (pod <- executorPodsWithIPs) {
+    for ((_, pod) <- executorPodsByIPs) {
```

Comment:
Per Javadocs at https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html

The following in my understanding.
ConcurrentHashMap is fully interoperable with Hashtable in programs that rely on its thread safety but not on its synchronization details. When you iterate, you are guaranteed that you get access to the all the entries in a thread-safe manner. Effectively, you are iterating on a snapshot of the Map's contents (and what you end up reading is dependent on concurrent modifications on the Map by other threads). Multiple threads can iterate on the HashMap in a thread-safe manner too as long as each thread has it own copy of the iterator. 

The method here only seems to require, thread-safe access to the map, and it seems to me the change should be safe.